### PR TITLE
12.0 afc 673

### PR DESCRIPTION
**Pull request by SebastianGieser-Vertel**

### CODE REVIEW

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+class CalendarOccasion(models.Model):
+    _inherit = "calendar.appointment"
+
+    xmlid_module="__adoxa_import__"
+
+    @api.model
+    def create_occasions(self):     
+        headers_header = ['occasions.csv', 'Notering',  'Trans', 'Odoo']
+        #path = os.path.join(config.options.get('data_dir'), 'Adoxa/occasions.csv')
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:46]: Remove commented out code, please

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+        path = "/usr/share/odoo-af/af_data_adoxa_loader/data/test_dumps/occasions.csv" #testing purposes only
+        header_path = "/usr/share/odoo-af/af_data_adoxa_loader/data/occasion_mapping.csv"
+        self.create_calendar_objs(headers_header, path, header_path)
+
+    @api.model
+    def create_appointments(self):     
+        headers_header = ['appointments.csv', 'Notering',  'Trans', 'Odoo']
+        #path = os.path.join(config.options.get('data_dir'), 'Adoxa/appointments.csv')
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:54]: Remove commented out code, please

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+    def create_calendar_objs(self, headers_header, path, header_path):
+        header_reader = ReadCSV(header_path, headers_header)
+        header_rows = header_reader.parse_header()
+        old_header = []
+        field_map = {}
+        transformations = {}
+        for row in header_rows:
+            
+            #_logger.info("header_rows row processing: %s" % row)
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:67]: Make it a debug-log?

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+            if row['Odoo'] != '' and "!" not in row['Odoo']:
+                field_map.update({row['Odoo']: row[headers_header[0]]})
+            # if row['Odoo2'] != '' and "!" not in row['Odoo2']:
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:70]: Remove commented out code, please

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+            #     field_map.update({row['Odoo2']: row[headers_header[0]]}) 
+            if row['Trans'] != '':
+                #_logger.info("transformations %s"%row['Trans'])
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:73]: Remove commented out code, please

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+                key = row['Trans'].split(",")[0]
+                value = row['Trans'].split(",")[1]
+                transformations.update({key: value})
+            old_header.append(row[headers_header[0]]) #adoxa fields
+        #_logger.info("old_header: %s" % old_header)
+        reader = ReadCSV(path, old_header)
+        iterations = 0
+        #_logger.info("get_data: %s" % next(reader.get_data()))
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:81]: Remove commented out code, please

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+        #reader.seek_zero()
+        for row in reader.get_data():
+            #_logger.info("row: %s" % row)
+            r = {}
+            header = list(field_map.keys())
+            #_logger.info("header: %s" % header)
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:87]: Make debug log?

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+            for i in range(len(header)):
+                if header[i] in field_map:
+                    #_logger.info("header %s: %s" % (i, row[field_map[header[i]]]))
+                    r.update({header[i] : row[field_map[header[i]]]})
+            #_logger.info("creating row %s" %r)
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:92]: Make debug log?

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+            self.create_occasion_from_row(r, transformations)
+            iterations += 1
+            if iterations > 500:
+                self.env.cr.commit()
+               #_loger.info("commit")
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:97]: Make debug log?

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+                iterations = 0
+        reader.close()
+        
+    
+    @api.model
+    def create_occasion_from_row(self, row, transformations):
+        #TODO: läs given address först, skapa den och sen lägg på fältet given_address_id = xmlid_to_res_id
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:104]: Is this a TODO, really? Adress has nothing to do with this?

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+                
+        #_logger.info("row: %s" % row)
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:106]: Remove commented out code, please

**Comment by MickeHolm** [comment]
Remove all commented out code and convert some of the commented out _logger.info to actual _logger.debugs

**Review by MickeHolm — CHANGES_REQUESTED** [review]

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 12]
```diff
@@ -0,0 +1,278 @@
...
+class CalendarOccasion(models.Model):
+    _inherit = "calendar.appointment"
+
+    xmlid_module="__adoxa_import__"
+
+    @api.model
+    def create_occasions(self):     
+        headers_header = ['occasions.csv', 'Notering',  'Trans', 'Odoo']
+        path = os.path.join(config.options.get('data_dir'), 'Adoxa/occasions.csv')
+        path = "/usr/share/odoo-af/af_data_adoxa_loader/data/test_dumps/occasions.csv" #testing purposes only
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:43]: path is set twice, why?

File: af_data_adoxa_loader/models/calendar_appointment.py [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+        header_path = "/usr/share/odoo-af/af_data_adoxa_loader/data/occasion_mapping.csv"
+        self.create_calendar_objs(headers_header, path, header_path)
+
+    @api.model
+    def create_appointments(self):     
+        headers_header = ['appointments.csv', 'Notering',  'Trans', 'Odoo']
+        path = os.path.join(config.options.get('data_dir'), 'Adoxa/appointments.csv')
+        path = "/usr/share/odoo-af/af_data_adoxa_loader/data/test_dumps/appointments.csv" #testing purposes only
```
Comment (MickeHolm) [inline af_data_adoxa_loader/models/calendar_appointment.py:51]: path is set twice, why?

**Review by MickeHolm — CHANGES_REQUESTED** [review]

File: af_data_adoxa_loader/models/calendar_appointment.py (hunk 12 above)
Comment (SebastianGieser-Vertel) [reply af_data_adoxa_loader/models/calendar_appointment.py:43]: one is for easy testing, the other for production

File: af_data_adoxa_loader/models/calendar_appointment.py (hunk 13 above)
Comment (SebastianGieser-Vertel) [reply af_data_adoxa_loader/models/calendar_appointment.py:51]: same as above

File: af_data_adoxa_loader/models/calendar_appointment.py (hunk 12 above)
Comment (MickeHolm) [reply af_data_adoxa_loader/models/calendar_appointment.py:43]: How do we switch to the production one? 
A new commit that removes the test one?

**Pull request merged**